platform-emf/src/main/java/com/softicar/platform/emf/table/configuration/EmfTableConfiguration.java

```diff
@@ -503,9 +505,27 @@ private void customizeAttributesAndOrder(EmfAttributeList<R, P> attributes) {
 
 	public void setBusinessKey(IDbKey<R> businessKey) {
 
+		checkBusinessKey(businessKey);
 		this.businessKey = businessKey;
 	}
 
+	private void checkBusinessKey(IDbKey<R> businessKey) {
+
+		if (!businessKey.isUniqueKey()) {
+			throw new SofticarDeveloperException(
+				"The set business key (%s) of table %s must be a unique key.",
+				Imploder.implode(businessKey.getFields(), " & "),
+				table);
+		}
+
+		if (table.getPrimaryKey().isGenerated() && !Collections.disjoint(businessKey.getFields(), table.getPrimaryKey().getFields())) {
+			throw new SofticarDeveloperException(
+				"The set business key (%s) of table %s must not contain an autoincrement column.",
```

Comment:
```suggestion
				"The business key (%s) of table %s must not contain a column of the generated primary key.",
```
we should not imply that we have a autoincrement column, that might change in the future